### ISSUE TITLE
ModuleNotFoundError for Matplotlib 

### ISSUE THREAD

**Issue by tpatpa** [yeatmanlab/pyAFQ#1028]
When attempting to use PyAFQ version 1.1 on Python version 3.8.18, I am encountering an unexpected error: `ModuleNotFoundError: No module named 'matplotlib'`

The PyAFQ version 1.1 should import successfully without any errors on Python 3.8.18 but a ModuleNotFoundError is raised for Matplotlib even though Matplotlib is not a required dependency for PyAFQ.

Installing Matplotlib version 3.7.3 allows successful importing of the PyAFQ package.

**Comment by arokem** [comment]
Hey @tpatpa : thanks for reporting this! Could you perhaps supply the full stack trace? Where in the code is this being raised? 

FWIW, I wouldn't mind making matplotlib a dependency, so if we find that this is in some place where we don't want to work around it, we can add that to the required depenedencies.

**Comment by arokem** [comment]
My guess is probably this: https://github.com/yeatmanlab/pyAFQ/blob/5e5df6261be69291abcad5249ff430961dae7616/AFQ/api/participant.py#L10

**Comment by tpatpa** [comment]
I have installed PyAFQ version 1.1 (`pip install pyafq==1.1`) on a fresh anaconda environment with Pyhton version 3.8.18 and simply tried importing the package `from AFQ.api.group import GroupAFQ`.

See more details:

```

{
	"name": "ModuleNotFoundError",
	"message": "No module named 'matplotlib'",
	"stack": "---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[3], line 1
----> 1 from AFQ.api.group import GroupAFQ

File ~/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/group.py:10
      <a href='file:///Users/m161121/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/group.py?line=7'>8</a> import logging
      <a href='file:///Users/m161121/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/group.py?line=8'>9</a> import AFQ.data.s3bids as afs
---> <a href='file:///Users/m161121/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/group.py?line=9'>10</a> from AFQ.api.participant import ParticipantAFQ
     <a href='file:///Users/m161121/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/group.py?line=10'>11</a> from AFQ.api.utils import (
     <a href='file:///Users/m161121/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/group.py?line=11'>12</a>     check_attribute, AFQclass_doc,
     <a href='file:///Users/m161121/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/group.py?line=12'>13</a>     export_all_helper, valid_exports_string)
     <a href='file:///Users/m161121/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/group.py?line=13'>14</a> import AFQ.utils.streamlines as aus

File ~/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/participant.py:10
      <a href='file:///Users/m161121/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/participant.py?line=7'>8</a> import tempfile
      <a href='file:///Users/m161121/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/participant.py?line=8'>9</a> import math
---> <a href='file:///Users/m161121/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/participant.py?line=9'>10</a> from matplotlib import font_manager
     <a href='file:///Users/m161121/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/participant.py?line=11'>12</a> from PIL import Image, ImageDraw, ImageFont
     <a href='file:///Users/m161121/opt/anaconda3/envs/DWI_newPython/lib/python3.8/site-packages/AFQ/api/participant.py?line=13'>14</a> from AFQ.definitions.mapping import SlrMap

ModuleNotFoundError: No module named 'matplotlib'"
}

```

**Comment by arokem** [comment]
Yep - that confirms my suspicion. Thanks again for reporting. I think that this issue will be addressed once we merge #1029 and make a new release, which we probably will quite soon, given the rapid accumulation of fixes and new features on the current release.

**Comment by tpatpa** [comment]
Happy to help and excited to try the new version!
